File: packages/registro/src/usecases/DeletarRegistro.ts
import { CasoDeUso } from "common";
import { BackendRegistro } from "../provider";

type Entrada = { registro_id: string };

export class DeletarRegistro implements CasoDeUso<Entrada, void> {
  public constructor(private backend: BackendRegistro) {}
  async executar(dto: Entrada): Promise<void> {
    await this.backend.deletar(dto.registro_id);
  }
}

File: apps/backend-springboot/src/main/java/br/com/dinheironapalmadamao/backend_springboot/core/registro/usecases/DeletarRegistro.java
package br.com.dinheironapalmadamao.backend_springboot.core.registro.usecases;

import java.util.concurrent.CompletableFuture;

import br.com.dinheironapalmadamao.backend_springboot.core.common.CasoDeUso;
import br.com.dinheironapalmadamao.backend_springboot.core.registro.provider.BackendRegistro;

public class DeletarRegistro implements CasoDeUso<Entrada, Void> {
  private BackendRegistro backend;

  public DeletarRegistro(BackendRegistro backend) {
    this.backend = backend;
  }

  @Override
  public CompletableFuture<Void> excutar(Entrada dto) {
    return CompletableFuture.runAsync(() -> {
      this.backend.deletar(dto.getRegistroId());
    });
  }

}
